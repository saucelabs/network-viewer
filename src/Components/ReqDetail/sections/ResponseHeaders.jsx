import React from 'react';
import PropTypes from 'prop-types';

import Styles from './../ReqDetail.styles.scss';

const ResponseHeaders = ({ data }) => (
  <div className={Styles['header-detail']}>
    {data.headers.response.map(({ name, value }, index) => (
      <p
        key={`${name}-${index}`}
        className={Styles['info-row']}
      >
        <span className={Styles['info-caption']}>
          {`${name}:`}
        </span>
        <span className={Styles['info-value']}>
          {value}
        </span>
      </p>
    ))}
  </div>
);

ResponseHeaders.propTypes = {
  data: PropTypes.object,
};

ResponseHeaders.defaultProps = {
  data: null,
};

export default ResponseHeaders;
