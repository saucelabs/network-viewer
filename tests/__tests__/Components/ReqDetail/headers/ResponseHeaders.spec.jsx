import React from 'react';
import { shallow } from 'enzyme';

import ResponseHeaders from '../../../../../src/Components/ReqDetail/headers/ResponseHeaders';

describe('ResponseHeaders', () => {
  const props = {
    data: {
      headers: {
        response: [{
          name: 'foo',
          value: 'bar',
        }],
      },
    },
    isURLEncoded: true,
  };

  it('renders without crashing', () => {
    const element = shallow(<ResponseHeaders {...props} />);
    expect(element).toMatchSnapshot();
  });
});
