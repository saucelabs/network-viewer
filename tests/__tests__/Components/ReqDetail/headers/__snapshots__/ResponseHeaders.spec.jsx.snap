// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponseHeaders renders without crashing 1`] = `
<div>
  <p
    key="foo-0"
  >
    <span>
      foo:
    </span>
    <span>
      bar
    </span>
  </p>
</div>
`;
